src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1118,6 +1135,235 @@ Containers::Optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id
+        case aiAnimBehaviour_CONSTANT:
+            return Animation::Extrapolation::Constant;
+        case aiAnimBehaviour_LINEAR:
+            return Animation::Extrapolation::Extrapolated;
+        case aiAnimBehaviour_REPEAT:
+            return Animation::Extrapolation::Constant;
+        default: CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+    }
+}
+
+}
+
+Containers::Optional<AnimationData> AssimpImporter::doAnimation(UnsignedInt id) {
+    /* Import either a single animation or all of them together. At the moment,
+       Blender doesn't really support cinematic animations (affecting multiple
+       objects): https://blender.stackexchange.com/q/5689. And since
+       https://github.com/KhronosGroup/glTF-Blender-Exporter/pull/166, these
+       are exported as a set of object-specific clips, which may not be wanted,
+       so we give the users an option to merge them all together. */
+    const std::size_t animationBegin =
+        configuration().value<bool>("mergeAnimationClips") ? 0 : id;
+    const std::size_t animationEnd =
+        configuration().value<bool>("mergeAnimationClips") ? _f->scene->mNumAnimations : id + 1;
+
+    /* Calculate total data size and track count. If merging all animations together,
+       this is the sum of all clip track counts. */
+    std::size_t dataSize = 0;
+    std::size_t trackCount = 0;
+    for (std::size_t a = animationBegin; a != animationEnd; ++a) {
+        const aiAnimation* animation = _f->scene->mAnimations[a];
+        for (std::size_t c = 0; c != animation->mNumChannels; ++c) {
+            const aiNodeAnim* channel = animation->mChannels[c];
+
+            /** @todo handle alignment once we do more than just four-byte types */
+            dataSize += channel->mNumPositionKeys*(sizeof(Float) + sizeof(Vector3)) +
+                channel->mNumRotationKeys*(sizeof(Float) + sizeof(Quaternion)) +
+                channel->mNumScalingKeys*(sizeof(Float) + sizeof(Vector3));
+
+            trackCount += (channel->mNumPositionKeys > 0 ? 1 : 0) +
+                (channel->mNumRotationKeys > 0 ? 1 : 0) +
+                (channel->mNumScalingKeys  > 0 ? 1 : 0);
+        }
+    }
+
+    /* Populate the data array */
+    Containers::Array<char> data{dataSize};
+
+    /* Import all tracks */
+    bool hadToRenormalize = false;
+    std::size_t dataOffset = 0;
+    std::size_t trackId = 0;
+    Containers::Array<Trade::AnimationTrackData> tracks{trackCount};
+    for(std::size_t a = animationBegin; a != animationEnd; ++a) {
+        const aiAnimation* animation = _f->scene->mAnimations[a];
+        for(std::size_t c = 0; c != animation->mNumChannels; ++c) {
+            const aiNodeAnim* channel = animation->mChannels[c];
+            const UnsignedInt target = doObject3DForName(channel->mNodeName.C_Str());
+            if(target == -1) {
+                Error{} << "Trade::AssimpImporter::animation(): unknown target node " << channel->mNodeName.C_Str();
```

Comment:
Minor thing (a test would catch this, I suppose) -- the Debug classes implicitly add spaces between printed values, so

```suggestion
                Error{} << "Trade::AssimpImporter::animation(): unknown target node" << channel->mNodeName.C_Str();
```

Reply:
Good catch. I tripped over this a few times before, probably won't be the last time 😄